pkg/controllers/subnetport/subnetport_controller.go

```diff
@@ -214,6 +214,18 @@ func (r *SubnetPortReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 		}
 		r.StatusUpdater.UpdateSuccess(ctx, subnetPort, setReadyStatusTrue, r.SubnetPortService)
 		if r.restoreMode {
+			// UpdateSuccess may fail due to k8s connection or update conflicts.
+			// In restore mode, we need to ensure the SubnetPort attachment Id is updated to the new SubnetPort before adding the annotation
+			// Otherwise VM operator will fail to get the new attachment ID.
+			updatedSubnetPort := &v1alpha1.SubnetPort{}
+			err := r.Client.Get(ctx, req.NamespacedName, updatedSubnetPort)
+			if err != nil {
+				return common.ResultNormal, err
+			}
+			if updatedSubnetPort.Status.Attachment.ID != subnetPort.Status.Attachment.ID {
+				log.Error(nil, "SubnetPort attachment ID is not updated, will retry later")
+				return common.ResultNormal, fmt.Errorf("SubnetPort Attachment ID is not updated")
```

Comment:
It looks that if it return error here, the reconciler has to get the subnetport and its state via NSX API again in the next reconciling. I feel that we'd better firstly add a retry in `.StatusUpdater.UpdateSuccess`, if the retry in `r.StatusUpdater.UpdateSuccess succeeds`, then we don't need to put it in the reconciler again, how do you think?